woodchipper/http/fastapi.py

```diff
@@ -79,5 +67,15 @@ async def send_with_extra_headers(message):
                 logging_ctx.update({"http.response.status_code": 500})
                 raise
 
+    def __build_middleware_stack__(self) -> ASGIApp:
+        asgi_app = self._app.__orig_build_middleware_stack__()
+        return type(self)(
+            asgi_app, request_id_factory=self._request_id_factory, blacklisted_headers=self._blacklisted_headers
+        )
+
     def chipperize(self):
-        self._app.build_middleware_stack = self._wrap_build_middleware_stack(self._app.build_middleware_stack)
+        if hasattr(self._app, "__orig_build_middleware_stack__"):
+            # This app has already been chipperized. No need to double up.
+            return
+        self._app.__orig_build_middleware_stack__ = self._app.build_middleware_stack
+        self._app.build_middleware_stack = self.__build_middleware_stack__
```

Comment:
You can arbitrarily set attributes of a FastAPI instance? Setting `__orig_build_middleware_stack__` doesn't attribute error?

Reply:
Correct.